### PR TITLE
More BOLT updates, to latest.

### PR DESCRIPTION
**Pull request by rustyrussell**
Includes the 12-block delay, removing non-htlc_maximum_msat updates, and setting the `dont_forward` bit!

Currently breaks lnprototest as expected, I'll have to fix that!

Closes: #5548 
Closes #5562

### CODE REVIEW

File: gossipd/routing.h [hunk 1]
```diff
@@ -399,21 +402,37 @@ bool routing_add_private_channel(struct routing_state *rstate,
...
+
+/**
+ * When a channel's funding has been spent.
+ */
+void routing_channel_spent(struct routing_state *rstate,
+			   u32 current_blockheight,
+			   struct chan *chan);
+
+/**
+ * Clean up and dying channels.
```
Comment (endothermicdev) [inline gossipd/routing.h:423]:
```suggestion
 * Clean up any dying channels.
```

File: doc/GOSSIP_STORE.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+This is placed in the gossip_store file when a funding transaction is
+spent.  `blockheight` is set to 12 blocks beyond the block containing
+the spend: at this point, gossipd will delete the channel.
+
+## Using the Gossip Store File
+
+- Always check the major version number!  We will increment it if the format
+  changes in a way that breaks readers.
+- Ignore unknown flags in the header.
+- Ignore unknown messages.
```
Comment (endothermicdev) [inline doc/GOSSIP_STORE.md:135]: I was reading this as "Ignore any message of unknown type." Is there any other validation a reader should be expected to perform? If so, we should be explicit.

**Review by endothermicdev — APPROVED** [review]
This was great - I learned a few things. I like the gossip_store versioning and new documentation.
ACK 6e4e104

**Comment by endothermicdev** [comment]
I should note this predictably upsets lnprototest during test_bolt7-01-channel_announcement-success.py.  Will need to bring the test up to speed with a 12 block delay after closing tx.

**Comment by cdecker** [comment]
Some things that made CI upset are fixed on `master` so a rebase will address at least the missing `mako` issue.

**Comment by cdecker** [comment]
Seems there is a mismatch between bolts and string in lnprototest:

```
        test += [
            Block(blockheight=103, number=3, txs=[funding_tx()]),
>           ExpectMsg(
                "channel_ready",
                channel_id=channel_id_v2(local_keyset),
                second_per_commitment_point=remote_per_commitment_point(1),
            ),
            # Ignore unknown odd messages
            TryAll([], RawMsg(bytes.fromhex("270F"))),
        ]

tests/test_bolt2-20-open_channel_accepter.py:1951: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = ExpectMsg:test_bolt2-20-open_channel_accepter.py:1951
msgtypename = 'channel_ready'
if_match = <function ExpectMsg._default_if_match at 0x7f5e32b7b9a0>
ignore = None, connprivkey = None
kwargs = {'channel_id': <function channel_id_v2.<locals>._channel_id_v2 at 0x7f5e302e5510>, 'second_per_commitment_point': functools.partial(<function remote_per_commitment_point.<locals>._remote_per_commitment_point at 0x7f5e302e55a0>, 1)}

    def __init__(
        self,
        msgtypename: str,
        if_match: Callable[["ExpectMsg", Message, "Runner"], None] = _default_if_match,
        ignore: Optional[Callable[[Message], Optional[List[Message]]]] = None,
        connprivkey: Optional[str] = None,
        **kwargs: Union[str, Resolvable],
    ):
        super().__init__(connprivkey)
        self.msgtype = namespace().get_msgtype(msgtypename)
        if not self.msgtype:
>           raise SpecFileError(self, "Unknown msgtype {}".format(msgtypename))
E           lnprototest.errors.SpecFileError: (ExpectMsg:test_bolt2-20-open_channel_accepter.py:1951, 'Unknown msgtype channel_ready')
```

**Comment by rustyrussell** [comment]
OK, so this now depends on new lnprototest, which requires #5621...

File: doc/GOSSIP_STORE.md (hunk 2 above)
Comment (rustyrussell) [reply doc/GOSSIP_STORE.md]: No, that's it, so I made it more explicit.

**Comment by rustyrussell** [comment]
OK, I folded the #5621 into this (at a point *before* we add the 12 block delay), so this PR will actually pass.

**Comment by rustyrussell** [comment]
Trivial rebase for minor test conflict.

**Comment by cdecker** [comment]
Seems some of the blockheights are off by 12 blocks, so this seems to be either the wrong commit or we need to adjust heights in tests.

**Comment by rustyrussell** [comment]
> Seems some of the blockheights are off by 12 blocks, so this seems to be either the wrong commit or we need to adjust heights in tests.
> 
> I'll be ignoring failures in the lnprototest config in CI until this PR is merged.

Wtf... Ok, I'll try to get to my laptop today and figure out what I did.

I meant to only add the lnprototest changes :(

**Comment by rustyrussell** [comment]
Ah, it really was a conflict!  That multichan test did need fixing, done now.

**Pull request merged**